embedchain/embedchain.py

```diff
@@ -136,6 +138,9 @@ def add(
         if data_type in {DataType.DOCS_SITE}:
             self.is_docs_site_instance = True
 
+        if dry_run:
+            return documents
```

Comment:
could you make it return a dict of

```python
{
    "chunks": {},
    "count": 0,
    "type": DataType
}
```
I mean count can obviously be deduced, but the type would be quite helpful.

Reply:
possibly meta_data too?

Reply:
```
def load_and_embed(self, loader: BaseLoader, chunker: BaseChunker, src, metadata=None, source_id=None):
        """
        Loads the data from the given URL, chunks it, and adds it to database.

        :param loader: The loader to use to load the data.
        :param chunker: The chunker to use to chunk the data.
        :param src: The data to be handled by the loader. Can be a URL for
        remote sources or local content for local loaders.
        :param metadata: Optional. Metadata associated with the data source.
        :param source_id: Hexadecimal hash of the source.
        :return: (List) documents (embedded text), (List) metadata, (list) ids, (int) number of chunks
        """
        embeddings_data = chunker.create_chunks(loader, src)

        # spread chunking results
        documents = embeddings_data["documents"]
        metadatas = embeddings_data["metadatas"]
        ids = embeddings_data["ids"]
```

If metadata and documents both are needed then we can directly use `embeddings_data` and drop ids from it. And in the same function we can get count which is returned as `count_new_chunks`  and ya data_type can be taken from the func param itself. So let me know if we can use `embeddings_data` directly

Reply:
wouldn't count `count_new_chunks` actually require for the embeddings to be added (i.e. not a dry run)? Couldn't we just count the list items?

Reply:
Ok then will use length of documents

Reply:
And I had one more doubt @cachho.
We need to return `embedding_data` which contains documents and metadatas in `load_and_embed` function before actually pushing it to db right?

Reply:
can we add an early return if it's a dry run?

Reply:
Yes or else it will be pushed into the db and then will get the documents.

Reply:
pushing to the db is not an option. please add an early return.